Docs: Add 8.2 implicit dependency migration (#28574)
storybookjs/storybook: MIGRATION.md

## From version 8.1.x to 8.2.x

### Failed to resolve import "@storybook/X" error

Storybook's package structure changed in 8.2. It is a non-breaking change, but can expose missing project dependencies.

> This happens when `@storybook/X` is missing in your `package.json`, but your project references `@storybook/X` in your source code (typically in a story file or in a `.storybok` config file). This is a problem with your project, and if it worked in earlier versions of Storybook, it was purely accidental.
> 
> Now in Storybook 8.2, it no longer works. The workaround is to install `@storybook/X` as a dev dependency and re-run.
> 
> Example errors:
> 
> - "Cannot find module @storybook/preview-api or its corresponding type declarations"
> - "Internal server error: Failed to resolve import "@storybook/theming/create" from ".storybook/theme.ts". Does the file exist?"

Reviewer: ````suggestion
This happens when `@storybook/X` is missing in your `package.json`, but your project references `@storybook/X` in your source code (typically in a story file or in a `.storybook` config file). This is a problem with your project, and if it worked in earlier versions of Storybook, it was purely accidental.

Now in Storybook 8.2, that incorrect project configuration no longer works. The solution is to install `@storybook/X` as a dev dependency and re-run.

Example errors:

```sh
Cannot find module @storybook/preview-api or its corresponding type declarations
```
```sh
Internal server error: Failed to resolve import "@storybook/theming/create" from ".storybook/theme.ts". Does the file exist?
```
````
